### DISCUSSION TITLE
d3 scaleBand paddingInner combination with paddingOuter(0)

### DISCUSSION THREAD

**Discussion by mitu-lv** [d3/d3#3946] [Q&A]
Sometimes I wish this would work.

```
const sb = d3.scaleBand()
  .domain([1, 2, 3])
  .range([0, 58])
  .paddingOuter(0)
  .paddingInner(0.4827586206896552)
  .align(0.5)
  .round(false);

sb.bandwidth() === 11.917808219178081; // true

58 - 11.917808219178081 * 3 = 58 - 35.75342465753424 = 22.24657534246576;

22.24657534246576 / 58 = 0.3835616438356166; // should be equal to inner padding

0.3835616438356166 !== 0.4827586206896552 // true


// Expected
58 * 0.4827586206896552 = 28;

28 / (3 - 1) = 14;

should be ok with padding as 14 and bandwidth 10
10 + 14 + 10 + 14 + 10 = 58

Seems that approach with step in calculation shifts the padding :(
```

**Comment by Fil** [comment]
The width (discounting `paddingOuter`) is divided between _n_ bands and _n-1_ pads; and the `bandwidth : padwidth` ratio is `(1-paddingInner) : paddingInner`.

With a width of 58 pixels and a bandwidth of 10 (padwidth = 14) this gives
`paddingInner = n - 58 / ((58 - 10) / (n - 1)) = 7/12 = 0.583333…`

See https://d3js.org/d3-scale/band

**Reply by mitu-lv** [reply]
That padding was 14 and bandwidth is 10 in my example. I have seen the source code for scaleBand. The part that I don't like is that numbers really fit with this combination of padding (0.4827586206896552) and length in pixels (58px), but scaleBand changes the padding size while it could still fit in those numbers. This is the result of having the step which of course works to fit everything inside, but with the expense to change input padding which still would be better than resulting padding after step calculation.

**Reply by Fil** [reply]
I fixed my comment, sorry for the inversion.
